services/common/src/redux/slices/complianceReportsSlice.ts

```diff
@@ -3,129 +3,197 @@ import { ENVIRONMENT } from "@mds/common/constants/environment";
-                thunkApi.dispatch(hideLoading());
-                return resp.data;
-            },
-            {
-                fulfilled: (state: ComplianceReportState, action) => {
-                    const records: IMineReportDefinition[] = action.payload.records ?? [];
-                    const { current_page, items_per_page, total, total_pages } = action.payload;
-                    state.reportPageData = { records, current_page, items_per_page, total, total_pages };
-                    state.params = action.meta.arg;
-                },
-                rejected: (state: ComplianceReportState, action) => {
-                    rejectHandler(action);
-                    // don't show loading forever if there's an error
-                    state.params = action.meta.arg;
-                }
-            }
-        )
-    }),
-    selectors: {
-        getMineReportDefinitionHash: (state): ItemMap<IMineReportDefinition> => {
-            return createItemMap(state.reportPageData.records, "mine_report_definition_guid")
+  name: complianceReportReducerType,
+  initialState,
+  reducers: (create) => ({
+    fetchComplianceReports: create.asyncThunk(
+      async (searchParams: ComplianceReportParams, thunkApi) => {
+        const headers = createRequestHeader();
+        thunkApi.dispatch(showLoading());
+        const resp = await CustomAxios({
+          errorToastMessage: "Failed to load compliance reports",
+        }).get(`${ENVIRONMENT.apiUrl}${API.MINE_REPORT_DEFINITIONS(searchParams ?? {})}`, headers);
+        thunkApi.dispatch(hideLoading());
+        return resp.data;
+      },
+      {
+        fulfilled: (state: ComplianceReportState, action) => {
+          const records: IMineReportDefinition[] = action.payload.records ?? [];
+          const { current_page, items_per_page, total, total_pages } = action.payload;
+          state.reportPageData = { records, current_page, items_per_page, total, total_pages };
+          state.params = action.meta.arg;
         },
-        getMineReportDefinitionOptions: (state): IMineReportDefinition[] => {
-            return state.reportPageData.records;
+        rejected: (state: ComplianceReportState, action) => {
+          rejectHandler(action);
+          // don't show loading forever if there's an error
+          state.params = action.meta.arg;
         },
-        getComplianceReportPageData: (state): IPageData<IMineReportDefinition> => {
-            return state.reportPageData;
+      }
+    ),
+    createMineReportDefinition: create.asyncThunk(
+      async (reportDefinition, thunkApi) => {
+        const headers = createRequestHeader();
+        thunkApi.dispatch(showLoading());
+        try {
+          const resp = await CustomAxios().post(
+            `${ENVIRONMENT.apiUrl}${API.MINE_REPORT_DEFINITION}`,
+            reportDefinition,
+            headers
+          );
+          return resp.data;
+        } catch (error) {
+          throw error;
```

Comment:
This block is a bit odd. I think you can take out the try-catch (`rejected` automatically catches, basically) and dispatch hideLoading before returning resp.data (does it work to return the function in the try block and then do something else in the finally?)